core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java

```diff
@@ -230,6 +238,7 @@ public void readFrom(StreamInput in) throws IOException {
         type = in.readString();
         id = in.readString();
         routing = in.readOptionalString();
+        parent = in.readOptionalString();
```

Comment:
Ignore this (and my other comments about version checks) if it is going to master only.